scipy/optimize/_minpack_py.py

```diff
@@ -304,9 +304,9 @@ def leastsq(func, x0, args=(), Dfun=None, full_output=0,
         A function or method to compute the Jacobian of func with derivatives
         across the rows. If this is None, the Jacobian will be estimated.
     full_output : bool, optional
-        non-zero to return all optional outputs.
+        If truthy, return all optional outputs (not just `x` and `ier`).
```

Comment:
Nit: when we specify the type of the argument as a `bool`, I would opt to talk about `True` and `False` rather than "truthy". This is a very old function, IIRC, that probably predated `True`'s addition to the language, hence the old-style language and default `full_output=0`. We can probably fix that to `full_output=False` with no problem.

```suggestion
        If ``True``, return all optional outputs (not just `x` and `ier`).
```

Reply:
Agreed. I updated the language.